src/utils/PreparedTodos.ts

```diff
@@ -0,0 +1,30 @@
+import { Todo } from '../types/Todo';
+import { TodosFilter } from '../types/TodoFilter';
+
+export const getpreparedTodos = (
+  todos: Todo[],
+  query: string,
+  filter: TodosFilter,
+) => {
+  let preparedTodos = [...todos];
+
+  if (query) {
+    preparedTodos = preparedTodos
+      .filter(({ title }) => title.toLowerCase().includes(query.toLowerCase()));
+  }
+
+  if (filter !== TodosFilter.All) {
+    switch (filter) {
+      case TodosFilter.Active:
+        preparedTodos = preparedTodos.filter((todo) => !todo.completed);
+        break;
+      case TodosFilter.Completed:
+        preparedTodos = preparedTodos.filter((todo) => todo.completed);
+        break;
+      default:
+        break;
+    }
+  }
+
+  return preparedTodos;
+};
```

Comment:
The `filter` method returns a filtered copy of the array, so we don't need to create it manually.
It makes sense to use the `switch` design when we have 3 or more conditions, or if we want to add an error to the `default` part, etc., otherwise it will be shorter to use `if`.
```suggestion
export const getpreparedTodos = (
  todos: Todo[],
  query: string,
  filterState: TodosFilter,
) => {
  if (query) {
    const preparedQuery = query.toLowerCase();
    todos = todos.filter(({ title }) => title.toLowerCase().includes(preparedQuery));
  }

  if (filterState === TodosFilter.Active) {
    todos = todos.filter((todo) => !todo.completed);
  } 
  if (filterState === TodosFilter.Completed) {
    todos = todos.filter((todo) => todo.completed);
  }

  return todos;
};
```